libs/api_common/device_helper.cpp

```diff
@@ -129,6 +129,10 @@ register_wait_async_ioctl_operation(_Inout_ async_ioctl_completion_t* async_ioct
 
     // Set the event on the thread-pool wait object.
     ebpf_assert(async_ioctl_completion->wait != nullptr);
+    if (async_ioctl_completion->wait == nullptr) {
```

Comment:
As Anurag pointed out in the other PR, why do we need both an assert and an 'if' test?  Seems like an assert would result in an analysis_assume rather than a runtime code check for something that cannot be hit in code coverage.